optimum/habana/transformers/generation/utils.py

```diff
@@ -1239,8 +1243,426 @@ def contrastive_search(
         >>> tokenizer.batch_decode(outputs, skip_special_tokens=True)
         ['DeepMind Company is a company that focuses on the development and commercialization of artificial intelligence (AI). DeepMind’s mission is to help people understand and solve problems that are difficult to solve in the world today.\n\nIn this post, we talk about the benefits of deep learning in business and how it']
         ```"""
+        # init values
```

Comment:
please specify the following code source link + what are the changes.

Reply:
@libinta I've added the code source link and changes